src/app/devices/sol/sol.component.ts

```diff
@@ -21,42 +31,67 @@ export class SolComponent implements OnInit {
   public server: string = environment.mpsServer.replace('https://', '')
   public uuid: string = ''
   public dataProcessor: any
-  public deviceState: number = 0
+  public powerState: any = 0
+  public isPoweredOn: boolean = false
+  public previousAction: string = 'sol'
+  public isLoading: boolean = false
 
-  constructor (private readonly activatedRoute: ActivatedRoute) { }
+  constructor (private readonly activatedRoute: ActivatedRoute, private readonly deviceService: DevicesService, public snackBar: MatSnackBar, public dialog: MatDialog) { }
 
   ngOnInit (): void {
     this.activatedRoute.params.subscribe(params => {
       this.uuid = params.id
     })
-    this.init()
-    if (this.redirector !== null) {
-      if (this.deviceState === 0) {
-        this.redirector.start(WebSocket)
-      }
-    }
-  }
-
-  ngAfterViewInit (): void {
-    this.term.open(this.container)
-    this.term.onData((data: any) => {
-      this.handleKeyPress(data)
-    })
-    this.term.attachCustomKeyEventHandler((e: any) => {
-      e.stopPropagation()
-      e.preventDefault()
-      if (e.ctrlKey && e.shiftKey && e.keyCode === 67) {
-        return navigator.clipboard.writeText(this.term.getSelection())
-      } else if (e.ctrlKey && e.shiftKey && e.keyCode === 86) {
-        return navigator.clipboard.readText().then(text => {
-          this.handleKeyPress(text)
+    this.setAmtFeatures()
+    this.deviceService.getPowerState(this.uuid).pipe(
+      catchError(err => {
+        console.log(err, 'coming error case')
+        this.snackBar.open($localize`Error enabling sol`, undefined, SnackbarDefaults.defaultError)
```

Comment:
Shouldnt this be an error related to getting the power state?

Reply:
done